packages/common/container-definitions/src/loader.ts

```diff
@@ -102,23 +102,134 @@ export interface ICodeAllowList {
 }
 
 /**
- * Events emitted by the Container "upwards" to the Loader and Host
+ * Events emitted by the {@link IContainer} "upwards" to the Loader and Host.
  */
+/* eslint-disable @typescript-eslint/unified-signatures */
 export interface IContainerEvents extends IEvent {
+    /**
+     * Emitted when the readonly state of the container changes.
+     *
+     * @remarks Listener parameters:
+     *
+     * - `readonly`: Whether or not the container is now in a readonly state.
+     *
+     * @see {@link IContainer.readOnlyInfo}
+     */
     (event: "readonly", listener: (readonly: boolean) => void): void;
+
+    /**
+     * Emitted when the {@link IContainer} completes connecting to the Fluid service.
+     *
+     * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+     *
+     * @see
+     *
+     * - {@link IContainer.connectionState}
+     *
+     * - {@link IContainer.connect}
+     */
     (event: "connected", listener: (clientId: string) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `codeDetails`: TODO
+     *
+     * - `proposal`: TODO
+     *
+     * @see {@link IContainer.proposeCodeDetails}
+     */
     (event: "codeDetailsProposed", listener: (codeDetails: IFluidCodeDetails, proposal: ISequencedProposal) => void);
+
+    /**
+     * TODO
```

Comment:
Legacy, not recommended for use.  Currently only fires during Container instantiation.

Reply:
We have a few usages in the repo. Do we have an alternate API existing consumers should migrate to?

Reply:
Those usages should really just be removed I think - the scenario they were intended to handle no longer exists.

I think our recommendation would just be to ignore it.  The prior guidance would have been to basically "reboot your app" when you see it, so now customers should just never have to do that.